### PR TITLE
Feature resolve variables cleanscript and selectionscript

### PR DESCRIPTION
**Pull request by pedrovelho**
Bring variable resolution at clean script on ThirdPartyCredentials, added test of variable resolution on CleanScript.

### CODE REVIEW

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [InternalTask.java#L387](https://github.com/ow2-proactive/scheduling/blob/fcfe9dc430815fbe22ad6bee038d302a3fbb9dfa/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/task/internal/InternalTask.java#L387): A "NullPointerException" could be thrown; "parent" is nullable here. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)
1. ![MAJOR][MAJOR] [InternalTask.java#L1179](https://github.com/ow2-proactive/scheduling/blob/fcfe9dc430815fbe22ad6bee038d302a3fbb9dfa/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/task/internal/InternalTask.java#L1179): A "NullPointerException" could be thrown; "obj" is nullable here. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/task/internal/InternalTask.java [hunk 1]
```diff
@@ -1222,6 +1223,8 @@ public synchronized void updateVariables(SchedulingService schedulingService) {
             }
 
             updatedVariables.putAll(getSystemVariables());
+
+            updatedVariables = VariableSubstitutor.resolveVariables(updatedVariables, updatedVariables);
```
Comment (jrochas) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/task/internal/InternalTask.java]: The two parameters of `resolveVariables` are identical. Is that ok?

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [InternalTask.java#L387](https://github.com/ow2-proactive/scheduling/blob/03a6c61daee60741733a484fa5850821b69dc46e/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/task/internal/InternalTask.java#L387): A "NullPointerException" could be thrown; "parent" is nullable here. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)
1. ![MAJOR][MAJOR] [InternalTask.java#L1179](https://github.com/ow2-proactive/scheduling/blob/03a6c61daee60741733a484fa5850821b69dc46e/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/task/internal/InternalTask.java#L1179): A "NullPointerException" could be thrown; "obj" is nullable here. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [InternalTask.java#L387](https://github.com/ow2-proactive/scheduling/blob/375f09c56e51f9fdeffc90f0a16659e293de0a76/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/task/internal/InternalTask.java#L387): A "NullPointerException" could be thrown; "parent" is nullable here. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)
1. ![MAJOR][MAJOR] [InternalTask.java#L1179](https://github.com/ow2-proactive/scheduling/blob/375f09c56e51f9fdeffc90f0a16659e293de0a76/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/task/internal/InternalTask.java#L1179): A "NullPointerException" could be thrown; "obj" is nullable here. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [InternalTask.java#L387](https://github.com/ow2-proactive/scheduling/blob/c576743c51bc79256620bd52e50610b87c41d2c1/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/task/internal/InternalTask.java#L387): A "NullPointerException" could be thrown; "parent" is nullable here. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)
1. ![MAJOR][MAJOR] [InternalTask.java#L1179](https://github.com/ow2-proactive/scheduling/blob/c576743c51bc79256620bd52e50610b87c41d2c1/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/task/internal/InternalTask.java#L1179): A "NullPointerException" could be thrown; "obj" is nullable here. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [InternalTask.java#L387](https://github.com/ow2-proactive/scheduling/blob/c33e9641d176d013dee23d4e2f124ad8c34726b4/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/task/internal/InternalTask.java#L387): A "NullPointerException" could be thrown; "parent" is nullable here. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)
1. ![MAJOR][MAJOR] [InternalTask.java#L1179](https://github.com/ow2-proactive/scheduling/blob/c33e9641d176d013dee23d4e2f124ad8c34726b4/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/task/internal/InternalTask.java#L1179): A "NullPointerException" could be thrown; "obj" is nullable here. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Pull request merged**